src/main/java/it/pagopa/transactions/services/v1/TransactionsService.java

```diff
@@ -498,60 +499,71 @@ public Mono<RequestAuthorizationResponseDto> requestTransactionAuthorization(
-                                                                                    .map(
-                                                                                            t -> new TransferListItemDto()
-                                                                                                    .creditorInstitution(
-                                                                                                            t.getPaFiscalCode()
-                                                                                                    )
-                                                                                                    .digitalStamp(
-                                                                                                            t.getDigitalStamp()
-                                                                                                    )
-                                                                                                    .transferCategory(
-                                                                                                            t.getTransferCategory()
-                                                                                                    )
-                                                                                    ).toList()
-                                                                    )
-                                                                    .isAllCCP(
-                                                                            transactionsUtils.isAllCcp(transaction, 0)
-                                                                    ),
-                                                            Integer.MAX_VALUE
-                                                    )
-                                                    .map(
-                                                            calculateFeeResponseDto -> Tuples.of(
-                                                                    calculateFeeResponseDto,
-                                                                    paymentSessionData
-                                                            )
-                                                    )
+                                            paymentSessionData -> {
+                                                String cardBin = switch (paymentSessionData) {
+                                                    case PaymentSessionData.CardSessionData cardSessionData ->
+                                                            cardSessionData.cardBin().value();
+                                                    case PaymentSessionData.PgsCardSessionData cardSessionData -> cardSessionData.cardBin().value();
+                                                    case PaymentSessionData.WalletCardSessionData cardSessionData ->
```

Comment:
`WalletCardSessionData`, `CardSessionData` and `PgsCardSessionData` could implement a common interface with methods: cardBin(), brand() and other common fields so that were needed those switch case can be collapsed to an if statement.
What do you think?